nemo_aligner/models/nlp/gpt/megatron_gpt_dpo_model.py

```diff
@@ -405,8 +405,13 @@ def finish_validation_step(self):
     @torch.no_grad()
     def get_logprob_batch(self, batch):
         seq_length = batch["chosen"].shape[1]
+        batch_size = batch["chosen"].shape[0]
 
-        data_iter = get_iterator_k_split(batch, get_num_microbatches())
+        assert (
+            batch_size * 2
+        ) % self.cfg.dpo.log_prob_forward_micro_batch_size == 0, f"batch_size ({batch_size}) * 2 is not divisble by log_prob_forward_micro_batch_size ({self.cfg.dpo.log_prob_forward_micro_batch_size})."
+        num_microbatches = int(batch_size * 2 // self.cfg.dpo.log_prob_forward_micro_batch_size)
+        data_iter = get_iterator_k_split(batch, num_microbatches)
```

Comment:
can you use the megatron core divide function? https://github.com/NVIDIA/Megatron-LM/blob/6dd3a1afa4e26d4d27e58d1e83aaa6ee6e36b477/megatron/core/utils.py#L34 otw lgtm

Reply:
Thanks for the pointer. Just updated.